lib/hammer_cli_foreman/compute_resources.rb

```diff
@@ -0,0 +1,43 @@
+require '../hammer-cli-foreman/lib/hammer_cli_foreman/compute_resource/base'
+Dir['../hammer-cli-foreman/lib/hammer_cli_foreman/compute_resource/*'].each {|file| require file }
+
+module HammerCLIForeman
+  module ComputeResourcesNew
+
+  end
+  def self.compute_resources
+    @compute_resources = {
+        'ec2' => HammerCLIForeman::ComputeResources::EC2.new,
```

Comment:
This way each time you call `HammerCLIForeman.compute_resources` all the resources are re-created.
It would be better if there is HammerCLIForeman.register_compute_resource(name, compute_resource) which is called from each of the compute resource files to register the new resource. Another benefit of this is that if you add new resource from e.g. plugin, you don't need to modify this method.